sdks/python/apache_beam/runners/direct/test_stream_impl.py

```diff
@@ -270,9 +273,20 @@ def events_from_rpc(endpoint, output_tags, coder):
     event_request = beam_runner_api_pb2.EventsRequest(
         output_ids=[str(tag) for tag in output_tags])
 
-    event_stream = stub.Events(event_request)
-    for e in event_stream:
-      yield _TestStream.test_stream_payload_to_events(e, coder)
+    event_stream = stub.Events(event_request, timeout=30)
+    try:
+      while True:
+        yield _TestStream.test_stream_payload_to_events(
+            next(event_stream), coder)
+    except StopIteration:
+      return
+    except grpc.RpcError as e:
+      if e.code() == grpc.StatusCode.DEADLINE_EXCEEDED:
```

Comment:
Have you tested this? I see that we're raising `RuntimeError` in streaming_cache.py, but here we're catching `grpc.RpcError` - is that as intended?

Reply:
Yes, this exception is specifically for GRPC timing out. Any runtime error on the server side gets caught by its exception handler (see test_stream_service.py)